### PR TITLE
Fixing uap serialization errors for exceptions

### PR DESCRIPTION
**Pull request by ViktorHofer**
Fixes https://github.com/dotnet/corefx/issues/23547

Exceptions are reflection blocked in uapaot.

### CODE REVIEW

File: src/System.Runtime.Serialization.Formatters/tests/EqualityExtensions.cs [hunk 1]
```diff
@@ -908,13 +908,15 @@ public static bool IsEqual(this Exception @this, Exception other)
             return @this != null &&
                 other != null &&
                 // On full framework, line number may be method body start
-                (PlatformDetection.IsFullFramework ? true :
+                // On Net Native we can't reflect on Exceptions and change its StackTrace
```
Comment (danmosemsft) [inline src/System.Runtime.Serialization.Formatters/tests/EqualityExtensions.cs]: nit, this whole expression is pretty big, maybe break it out into pieces

File: src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterHelper.cs [hunk 2]
```diff
@@ -136,7 +136,10 @@ private static void SanityCheckBlob(object obj, string[] blobs)
                 // different blobs at runtime for some obscure reason. Excluding those from the check.
                 !(obj is WeakReference<Point>) &&
                 !(obj is Collections.Specialized.HybridDictionary) &&
-                !(obj is TimeZoneInfo.AdjustmentRule))
+                // Temporary disabled until TimeZoneInfo is converged. Issue #23804
```
Comment (danmosemsft) [inline src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterHelper.cs]: I'd make a note in #23804 to remove this

**Comment by ViktorHofer** [comment]
thanks!

File: src/System.Runtime.Serialization.Formatters/tests/EqualityExtensions.cs (hunk 1 above)
Comment (danmosemsft) [reply src/System.Runtime.Serialization.Formatters/tests/EqualityExtensions.cs]: you simplified the other one :) up to you

**Comment by ViktorHofer** [comment]
@dotnet-bot test NETFX x86 Release Build

**Pull request merged**